Missionframework/scripts/shared/functions/F_kp_checkGear.sqf

```diff
@@ -19,44 +19,214 @@
+                } else {
+                    _removed pushBack _optic;
+                }
+            };
+
+            _magazine params [["_magazineClass", ""], ["_magazineAmmoCount", 0]];
+            if (_magazineClass != "") then {
+                if (toLower _magazineClass in KP_liberation_allowed_items) then {
+                    _container addMagazineAmmoCargo [_magazineClass, 1, _magazineAmmoCount];
+                } else {
+                    _removed pushBack _magazineClass;
+                }
+            };
+
+            _magazineGL params [["_magazineGLClass", ""], ["_magazineGLAmmoCount", 0]];
+            if (_magazineGLClass != "") then {
+                if (toLower _magazineGLClass in KP_liberation_allowed_items) then {
+                    _container addMagazineAmmoCargo [_magazineGLClass, 1, _magazineGLAmmoCount];
+                } else {
+                    _removed pushBack _magazineGLClass;
+                }
+            };
+        } else {
+            _removed pushBack _weapon;
+        }
+    } forEach _weaponsItemsCargo;
+
+    _removed;
+};
+
+_removedItems append ([uniformContainer player] call _removeWeapons);
+_removedItems append ([vestContainer player] call _removeWeapons);
+_removedItems append ([backpackContainer player] call _removeWeapons);
+
+private _weapons = weapons player;
+_weapons = _weapons apply { toLower _x };
+{
+    player removeWeapon _x;
+    _removedItems pushBack _x;
+} forEach (_weapons - KP_liberation_allowed_items);
+
+private _weaponItems = primaryWeaponItems player;
+_weaponItems append primaryWeaponMagazine player;
+_weaponItems = _weaponItems apply { toLower _x };
+{
+    player removePrimaryWeaponItem _x;
+    _removedItems pushBack _x;
+} forEach (_weaponItems - KP_liberation_allowed_items);
+
+_weaponItems = secondaryWeaponItems player;
+_weaponItems append secondaryWeaponMagazine player;
+_weaponItems = _weaponItems apply { toLower _x };
+{
+    player removeSecondaryWeaponItem _x;
+    _removedItems pushBack _x;
+} forEach (_weaponItems - KP_liberation_allowed_items);
+
+_weaponItems = handgunItems player;
+_weaponItems append handgunMagazine player;
+_weaponItems = _weaponItems apply { toLower _x };
+{
+    player removeHandgunItem _x;
+    _removedItems pushBack _x;
+} forEach (_weaponItems - KP_liberation_allowed_items);
+
+_removedItems = (_removedItems arrayIntersect _removedItems) - [""];
+if (count _removedItems > 0) then {
+	private _text = format ["[KP LIBERATION] [BLACKLIST] Found %1 at Player %2", (_removedItems), name player];
 	_text remoteExec ["diag_log",2];
 	private _badItems = "";
-	{if (((_x find "acre") == -1) && ((_x find "tf_") == -1) && ((_x find "tfar_") == -1)) then {_badItems = _badItems + _x + "\n";};} forEach (_playerItems - KP_liberation_allowed_items);
+	{if (((_x find "acre") == -1) && ((_x find "tf_") == -1) && ((_x find "tfar_") == -1)) then {_badItems = _badItems + _x + "\n";};} forEach (_removedItems);
```

Comment:
Guess this is still in the work due to your comment concerning ACRE/TFAR.
But in the end you shouldn't need to fetch them in a new local variable and may just do (not tested, as at the office)
```sqf
hint format [localize "STR_BLACKLISTED_ITEM_FOUND", _removedItems joinString "\n"];
```

Reply:
I changed it to use `joinString`, good point there